Neos.ContentRepository.Core/Classes/Projection/ContentGraph/Nodes.php

```diff
@@ -124,6 +124,11 @@ public function merge(self $other): self
         return self::fromArray($nodes);
     }
 
+    public function append(Node $node): self
```

Comment:
Why not use merge?

Reply:
With merge I would expect an instance of `Nodes`. To add a single node to a set, "append" seems fine for me

Reply:
True, but it wouldn't be the first place we use a `merge(Nodes::fromArray([$node]))` kind of construct ("upcast" single to collection). I just wanted to throw the question out there if we want to go down this route, because IMHO then we should add append (or similar) to other places probably?

Reply:
In other places we named it more explicitly, like [NodeTypeNames::withAdditionalNodeTypeName()](https://github.com/neos/neos-development-collection/blob/88ccba7dd15042ba3a46f055a8f6ac5ac17b30b4/Neos.ContentRepository.Core/Classes/NodeType/NodeTypeNames.php#L63).
So we could call it `withAppendedNode()` for example.
But IMO `merge(Nodes::fromArray([$node]))` would be worse because it's
- harder to comprehend
- less explicit (with "append" it's clear that the new node is added to the end of the set)
- slower (OK, that is probably negligible)

Reply:
btw: I think `prepend()` would make more sense in this example

Reply:
Yep, was mostly arguing for remembering to do it everywhere like this :) I am fine with having it, and yeah in htis case `prepend` seems sensible

Reply:
Adjusted as requested :)

Reply:
Sorry for re-opening, but when I suggested "Prepend" what I actually meant was `withPrependedNode()` – but I'm not sure whether we use the `with`er construct consistently everywhere.

Reply:
ahh i see for collections the pattern as far as i have seen is to have `merge` or a method `union` and then i thought `append` as well but i get your point.